project-1/gspread_code.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Oct 16 15:12:29 2023
+
+@author: computer world
+
+Project : getting data from github repositories(through selenium library) and 
+sending to google spreadsheet with gspread package
+"""
+
+import gspread
+import pandas as pd
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+
+path = "documents/datasets_for_data_science/gspread_project/spartan-figure-402211-2241d0349aab.json"
+
+def update_spreadsheet(spreadsheet = "Gspread SpreadSheet",worksheet="Sheet1"):
+    
+    client = gspread.service_account(filename=path)
+    worksheet = client.open(spreadsheet).worksheet(worksheet)
+    worksheet.clear() #clearning worksheet on each run of program
+    
+    def github_data(url = "https://github.com/Humphery7?tab=repositories"):
```

Comment:
Why use a nested function here? Zen of python: "Simple is better than complex"

Reply:
Ohh alright i actually didn't realize it made it more complex
it just cause from youtube videos they do that sometimes and it made it look more cleaner.

but i will be looking out as to whether it is adding to the complexity now, i will change it

Reply:
Aside from functionality, readability is very important in programming. This piece will definitely work without the nested approach here unless you're thinking OOP, which you can also shoot at!